core/services/ocr2/plugins/ocr2keeper/evmregistry/v21/mercury/streams/streams.go

```diff
@@ -171,71 +171,103 @@ func (s *streams) buildResult(ctx context.Context, i int, checkResult ocr2keeper
 	// in the revert for mercury v0.2, which is denoted by time in the struct bc starting from v0.3, only timestamp will be supported
 	streamsLookupResponse.Block = uint64(block.Int64())
 	lookupLggr.Infof("at block %d upkeep %s DecodeStreamsLookupRequest feedKey=%s timeKey=%s feeds=%v time=%s extraData=%s", block, upkeepId, streamsLookupResponse.FeedParamKey, streamsLookupResponse.TimeParamKey, streamsLookupResponse.Feeds, streamsLookupResponse.Time, hexutil.Encode(streamsLookupResponse.ExtraData))
-	lookups[i] = streamsLookupResponse
+	lookups[len(lookups)] = streamsLookupResponse
```

Comment:
Not sure i follow this change. Won't this break indexing done in line 104? (when checkResults[i] is modified)

Reply:
This `buildResult ` func build `mercury.StreamsLookup` object and add to map. the map is keyed by an integer, which is basically the index or order in the map. 

After the map is populated, in L104, we use the key (i) and value (lookup object) to doLookup().  I dont think it breaks any indexing~

Reply:
what will happen if some of the checkResults don't need any lookup?